samza-core/src/main/scala/org/apache/samza/checkpoint/CheckpointTool.scala

```diff
@@ -158,14 +159,12 @@ class CheckpointTool(newOffsets: TaskNameToCheckpointMap, coordinatorStreamStore
     combinedConfigMap.putAll(userDefinedConfig)
     val combinedConfig: Config = new MapConfig(combinedConfigMap)
 
+    val taskConfig = new TaskConfig(combinedConfig)
     // Instantiate the checkpoint manager with coordinator stream configuration.
-    val checkpointManager: CheckpointManager = combinedConfig.getCheckpointManagerFactory() match {
-      case Some(className) =>
-        ReflectionUtil.getObj(classLoader, className, classOf[CheckpointManagerFactory])
-          .getCheckpointManager(combinedConfig, new MetricsRegistryMap)
-      case _ =>
-        throw new SamzaException("Configuration: task.checkpoint.factory is not defined.")
-    }
+    val checkpointManager: CheckpointManager =
+      JavaOptionals.toRichOptional(taskConfig.getCheckpointManager(new MetricsRegistryMap, classLoader))
```

Comment:
Minor: Can we just call Optional#orElseThrow instead?

Reply:
I tried that, but the Scala syntax for a 0-arg method isn't compatible with Java `Supplier`.
For example, in Scala, `val supplier: Supplier[String] = () => "hello world"` is a "type mismatch" error.